llvm/test/MC/Disassembler/AMDGPU/vinterp.txt

```diff
@@ -0,0 +1,723 @@
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, -v2, v3 wait_exp:0
+
+0x00,0x00,0x02,0xcd,0x01,0x05,0x0e,0x84
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, -v3.l wait_exp:0
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, -v3.l wait_exp:0
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, -v3 wait_exp:0
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, -v3 wait_exp:0
+
+0x00,0x80,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l clamp wait_exp:0
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l clamp wait_exp:0
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 clamp wait_exp:0
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 clamp wait_exp:0
+
+0x00,0x01,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l wait_exp:1
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l wait_exp:1
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 wait_exp:1
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 wait_exp:1
+
+0x00,0x07,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l wait_exp:7
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l wait_exp:7
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 wait_exp:7
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 wait_exp:7
+
+0x00,0x08,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.h, v2, v3.l op_sel:[1,0,0,0] wait_exp:0
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.h, v2, v3.l op_sel:[1,0,0,0] wait_exp:0
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[1,0,0,0] wait_exp:0
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[1,0,0,0] wait_exp:0
+
+0x00,0x10,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l op_sel:[0,1,0,0] wait_exp:0
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l op_sel:[0,1,0,0] wait_exp:0
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[0,1,0,0] wait_exp:0
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[0,1,0,0] wait_exp:0
+
+0x00,0x20,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.h op_sel:[0,0,1,0] wait_exp:0
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.h op_sel:[0,0,1,0] wait_exp:0
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[0,0,1,0] wait_exp:0
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[0,0,1,0] wait_exp:0
+
+0x00,0x40,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l op_sel:[0,0,0,1] wait_exp:0
+# GFX12-TRUE16: v_interp_p10_f16_f32 v0, v1.l, v2, v3.l op_sel:[0,0,0,1] wait_exp:0
+# GFX11-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[0,0,0,1] wait_exp:0
+# GFX12-FAKE16: v_interp_p10_f16_f32 v0, v1, v2, v3 op_sel:[0,0,0,1] wait_exp:0
+
+0x00,0x78,0x02,0xcd,0x01,0x05,0x0e,0x04
+# GFX11-TRUE16: v_interp_p10_f16_f32 v0, v1.h, v2, v3.h op_sel:[1,1,1,1] wait_exp:0
```

Comment:
These tests where op_sel is applied to 32-bit arguments do not make sense, as you should not apply op_sel to those. However, I see they have existed a while, and were just ported in this patch. And we generally are not strictly rejecting things in the disassembler.